### PR TITLE
[Runtime] add failurePolicy to runtime plugin

### PR DESCRIPTION
**Pull request by Binyang2014**
Refer to issue #4014 add failurePolicy for runtime plugin.
User can make runtime plugin doesn't fail the entire runtime by setting `failurePolicy: ignore`

The default failurePolicy is `fail`

### CODE REVIEW

File: src/kube-runtime/src/plugins/plugin_utils.py [hunk 1]
```diff
@@ -16,36 +16,41 @@
...
-
-    return [parameters, args.pre_script, args.post_script]
-
-def inject_commands(commands, script):
-    if commands is not None and len(commands) > 0:
-        new_commands = [x+"\n" for x in commands]
-        with open(script, 'a+') as f:
-            f.writelines(new_commands)
-
+    plugin_config = yaml.load(args.plugin_config, Loader=yaml.SafeLoader)
```
Comment (abuccts) [inline src/kube-runtime/src/plugins/plugin_utils.py]: unify to one of `safe_load` and `load(Loader=yaml.SafeLoader)`?

File: src/kube-runtime/src/plugins/plugin_utils.py [hunk 2]
```diff
@@ -16,36 +16,41 @@
...
+from common.utils import init_logger
+
+LOGGER = logging.getLogger(__name__)
+
+
+class PluginHelper:  #pylint: disable=too-few-public-methods
+    def __init__(self, plugin_config: dict):
+        self._parameters = plugin_config.get("parameters", "")
+        self._plugin_name = plugin_config.get("plugin", "")
+        self._failure_policy = plugin_config.get("failurePolicy", "fail")
```
Comment (abuccts) [inline src/kube-runtime/src/plugins/plugin_utils.py]: is there a validation for this field?

File: src/kube-runtime/src/plugins/plugin_utils.py (hunk 1 above)
Comment (Binyang2014) [reply src/kube-runtime/src/plugins/plugin_utils.py]: fixed

File: src/kube-runtime/src/plugins/plugin_utils.py (hunk 2 above)
Comment (Binyang2014) [reply src/kube-runtime/src/plugins/plugin_utils.py]: If this field is not `ignore`, we will treat it as `fail`.  Refer to https://github.com/microsoft/pai/pull/4048/files#diff-33c702285a3718b9aae88340cbe60538R38. Not validate it here, just trust rest-server will pass valid config to runtime.
Invalid config may fail the runtime and job.

**Pull request merged**